webapp/graphite/render/functions.py

```diff
@@ -109,6 +109,8 @@ def getNodeOrTag(series, n, pathExpression=None):
 
 
 def aggKey(series, nodes, pathExpression=None):
+  if series is None:
+    return None
```

Comment:
are you sure it won't lead to issues that now `aggKey()` may not return a string? for example in `aliasByNode()` it's going to assign that `None` to `series.name`: https://github.com/grafana/graphite-web/blob/65457e682a68b2c6671a856ab4ac66e56da18975/webapp/graphite/render/functions.py#L2531

Reply:
I'm not entirely sure no. I'm not too familiar with the code. Hmmm

Reply:
well in the case of aliasByNode, can `series` be none? I would expect not, no?

Reply:
It feels like I should return `""` instead of `None` but I'm not convinced it would prevent all the potential issues.

Reply:
I think it would be better to stick to a string, whether it's `""` or `"<missing>"` or whatever, but I think putting `None` there can cause other issues

Reply:
should I put the check at the calling place in weightedAverage instead?

Reply:
> I think it would be better to stick to a string, whether it's `""` or `"<missing>"` or whatever, but I think putting `None` there can cause other issues

ok

Reply:
I changed strategy: see 635fd83

Reply:
are you specifically just trying to fix an issue in `weightedAverage()`, do other functions not have the same problem of potentially passing a `None` into `aggKey()`?

Reply:
I'm trying to only fix weightedAverage. A quick look tells me its unlikely to happen with other functions. However I'm no graphite expert :/

Reply:
anyway, the fix for `weightedAverage()` should work I think. did you test it?

Reply:
yeah it's good :)